tests/test_dataset_upload.py

```diff
@@ -0,0 +1,101 @@
+import json
+from http.server import BaseHTTPRequestHandler
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import ClassVar, List
+
+from kagglehub.datasets import dataset_upload
+from kagglehub.gcs_upload import TEMP_ARCHIVE_FILE
+from tests.fixtures import BaseTestCase
+
+from .utils import create_test_http_server
+
+GET_DATASET = "/datasets/view/akankshaaa013/top-grossing-movies-dataset"
+CREATE_DATASET = "/datasets/create/new"
+
+TEMP_TEST_FILE = "temp-test-file"
+
+
+
+class KaggleAPIHandler(BaseHTTPRequestHandler):
```

Comment:
We are now using flask. This was the old style of writing test which relied on the base http package. Make sure to follow the new style. 

I see you added some `dataset_upload_stub.py`. Maybe you are actively working on that. If that's the case, you can disregard this comment.